app/UI/components/right-panel.tsx

```diff
@@ -0,0 +1,79 @@
+/** @jsx h */
+import { Component, createRef, h } from "https://esm.sh/preact@10.17.1";
+import { IconButtonClass } from "./tw.ts";
+import { CloseIcon } from "../icons/close-icon.tsx";
+import { ComponentChildren } from "https://esm.sh/preact@10.17.1";
+import { Channel } from "https://raw.githubusercontent.com/BlowaterNostr/csp/master/csp.ts";
+
+type RightPanelProps = {
+    inputChan: RightPanelInputChannel;
+};
+
+type RightPanelState = {
+    show: boolean;
+};
+
+export type RightPanelInputChannel = Channel<ComponentChildren>;
```

Comment:
Maybe we over engineered. Type alias in TS doesn't create a new type. Kinda unnecessary.